### ISSUE TITLE
Tapping a micromodal close link (data-micromodal-close) triggers click on links underneath the modal on iOS

### ISSUE THREAD

**Issue by kisabelle** [ghosh/Micromodal#324]
Doesn't seem to happen on desktop browsers, but on my iPhone 11 in Chrome, Safari and Firefox, when tapping on the close button the micromodal closes but then any link that was below the modal and directly under the close button will be clicked and followed as well.

My close button is marked up as follows and positioned absolutely in the top right corner of the modal:

`<a class="modal__close" role="button" aria-label="Close modal" data-micromodal-close></a>`

In one example, the button to close the modal sits directly above the button to open the navigation. The navigation button is triggered when closing the modal.

![Screen Shot 2020-05-08 at 3 56 35 PM](https://user-images.githubusercontent.com/1544652/81455849-4be08e00-9145-11ea-8cdc-65b969c71ec1.png)

![Screen Shot 2020-05-08 at 3 56 47 PM](https://user-images.githubusercontent.com/1544652/81455850-4c792480-9145-11ea-8e3a-c3ab281da7c2.png)

**Comment by kisabelle** [comment]
Update: I have just figured out this happens on another modal library as well. Could be a browser-related bug. If anyone has any ideas on how to prevent this behaviour, I would greatly appreciate it.

**Comment by kisabelle** [comment]
Update: Adding the following closeModal() fixed it:

```
event.preventDefault();
event.stopPropagation();
```

**Comment by Green-Zero-Development** [comment]
@kisabelle 
THANK YOU for figuring this out. It's been driving me crazy for a week or so now.

Your fix also worked for me.

**Comment by thetwopct** [comment]
@kisabelle could you share your solution? What I'm doing doesn't seem to solve this for me.

**Comment by badams2527** [comment]
If anyone has troubles with this, and they don't know where to put @kisabelle code, here was my solution:
```
MicroModal.init({
        disableFocus: true,
        onClose: function(modal, element, event) {
            event.preventDefault();
            event.stopPropagation();
        }
    });
```

**Comment by kisabelle** [comment]
@thetwopct I mentioned in my original post, you need to find the `closeModal()` function in `micromodal.js`. It starts around line 256

```
value: function closeModal() {
          var event = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : null;
          var modal = this.modal;
          this.modal.setAttribute('aria-hidden', 'true');
          this.removeEventListeners();
          this.scrollBehaviour('enable');

          if (this.activeElement && this.activeElement.focus) {
            this.activeElement.focus();
          }

          this.config.onClose(this.modal, this.activeElement, event);

          if (this.config.awaitCloseAnimation) {
            var openClass = this.config.openClass; // <- old school ftw

            this.modal.addEventListener('animationend', function handler() {
              modal.classList.remove(openClass);
              modal.removeEventListener('animationend', handler, false);
            }, false);
          } else {
            modal.classList.remove(this.config.openClass);
          }
          
          event.preventDefault();
          event.stopPropagation();
        }
```

**Comment by rutchkiwi** [comment]
@ghosh I believe that basically everyone using this library wants to have the solution given above by @kisabelle and @badams2527 built-in, whether they know it or not! Do you see any problem with including that as the default behavior? (other than the extra 30 or so bytes). I can raise a PR with these changes if needed.

**Comment by kisabelle** [comment]
@ghosh @rutchkiwi I would love to see this added – you've just reminded me that I've since installed Micromodal on a couple of other sites without including this fix!

**Comment by ptrsmk** [comment]
@ghosh hasn't made a contribution to any repo on github in about 3 months.  It may be time for a new maintainer.

**Comment by HongPong** [comment]
Thank you all, this solves my client issue and I hope it gets picked up as a PR #444  eventually. 

Just to note for this thread, that i changed /lib/src/index.js which is a little different than the dist build artifact file noted above. I've been stuck on this for a while so I truly appreciate it!!

**Comment by toastal** [comment]
This was an issue with me on Epiphany (GNOME Web) as well. The bug said wasn't an issue in Safari, but it would seem it could certainly be a Webkit issue.

**Comment by ghosh** [comment]
Closed in #479 and 79c75958c6aaa9cc00447b8a58ab7efbf2e84594
